build/resolve_buck.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+import os
+import platform
+import stat
+import sys
+import tempfile
+import urllib.request
+
+from dataclasses import dataclass
+from pathlib import Path
+
+import buck_util
+import zstd
+
+"""
+Locate or download the version of buck2 needed to build ExecuTorch.
+It is intended to be invoked from the CMake build logic, and it returns
+the path to 'buck2' via stdout. Log messages are written to stderr.
+
+It uses the following logic, in order of precedence, to locate or download
+buck2:
+
+ 1) If BUCK2 argument is set explicitly, use it. Warn if the version is
+    incorrect.
+ 2) Look for a binary named buck2 on the system path. Take it if it is
+    the correct version.
+ 3) Check for a previously downloaded buck2 binary (from step 4).
+ 4) Download and cache correct version of buck2.
+
+"""
+
+# Path to the file containing BUCK2 version (build date) for ExecuTorch.
+BUCK_VERSION_FILE = ".ci/docker/ci_commit_pins/buck2.txt"
```

Comment:
Please comment on what this is relative to. Below, consider loading it relative to this script's location, so that the current working directory doesn't change the behavior.